File: my_first_app/lib/src/features/saved_threads/domain/models/saved_thread.dart
class SavedThread {
  final int id;
  final String storyHnId;
  final String? title;
  final String? url;
  final String createdAt;
  final List<SavedThreadItem> items;

  const SavedThread({
    required this.id,
    required this.storyHnId,
    required this.title,
    this.url,
    required this.createdAt,
    required this.items,
  });
}

class SavedThreadItem {
  final String itemType;
  final String hnId;
  final String? rawText;
  final Map<String, dynamic>? aiSummary;
  final String? modelName;
  final String modelVersion;
  final String createdAt;

  const SavedThreadItem({
    required this.itemType,
    required this.hnId,
    required this.rawText,
    required this.aiSummary,
    required this.modelName,
    required this.modelVersion,
    required this.createdAt,
  });
}

File: hackernews/app/schemas/saved_thread.py
from datetime import datetime
from typing import List, Optional
from pydantic import BaseModel


class SavedThreadCreate(BaseModel):
    story_hn_id: int
    comment_hn_ids: List[int] = []


class SavedThreadQueueOut(BaseModel):
    status: str
    story_hn_id: int
    comment_count: int


class SavedThreadItemOut(BaseModel):
    item_type: str
    hn_id: int
    raw_text: Optional[str]
    ai_summary: Optional[dict]
    model_name: Optional[str]
    model_version: str
    created_at: datetime

    class Config:
        from_attributes = True


class SavedThreadOut(BaseModel):
    id: int
    story_hn_id: int
    title: Optional[str]
    url: Optional[str]
    created_at: datetime
    items: List[SavedThreadItemOut] = []

    class Config:
        from_attributes = True
